src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -399,20 +397,26 @@ private RowMapBuffer getTransactionRows(BinlogConnectorEvent beginEvent) throws
 				case QUERY:
 					QueryEventData qe = event.queryData();
 					String sql = qe.getSql();
+					String upperCaseSql = sql.toUpperCase();
 
-					if ( sql.toUpperCase().startsWith(BinlogConnectorEvent.SAVEPOINT)) {
+					if ( upperCaseSql.startsWith(BinlogConnectorEvent.SAVEPOINT)) {
 						LOGGER.debug("Ignoring SAVEPOINT in transaction: " + qe);
 					} else if ( createTablePattern.matcher(sql).find() ) {
 						// CREATE TABLE `foo` SELECT * FROM `bar` will put a CREATE TABLE
 						// inside a transaction.  Note that this could, in rare cases, lead
 						// to us starting on a WRITE_ROWS event -- we sync the schema position somewhere
 						// kinda unsafe.
 						processQueryEvent(event);
-					} else if (sql.toUpperCase().startsWith("INSERT INTO MYSQL.RDS_HEARTBEAT")) {
+					} else if (upperCaseSql.contains("MYSQL.RDS_")) {
```

Comment:
Feels a bit risky here, can we be more restrictive like `upperCaseSql.startsWith("INSERT INTO MYSQL.RDS_") || upperCaseSql.startsWith("DELETE FROM MYSQL.RDS_")` ?

Reply:
Also might be a good idea to move this check to be the last case since it's rare.

Reply:
👍 however I feel it happens more frequently than `DROP TEMPORARY TABLE` so I will leave the order unchanged.